### ISSUE TITLE
NullPointerException in DependencyUpdatesXmlRenderer.getVersionsBlocks with version ranges

### ISSUE THREAD

**Issue by TobiX** [mojohaus/versions-maven-plugin#323]
When there is a dependency with a version range (I know, bad practice) in the `dependencyManagement` the xml renderer throws a null pointer exception:

```
Caused by: java.lang.NullPointerException
    at org.codehaus.mojo.versions.DependencyUpdatesXmlRenderer.getVersionsBlocks (DependencyUpdatesXmlRenderer.java:190)
    at org.codehaus.mojo.versions.DependencyUpdatesXmlRenderer.getDependencyInfoBlock (DependencyUpdatesXmlRenderer.java:251)
    at org.codehaus.mojo.versions.DependencyUpdatesXmlRenderer.render (DependencyUpdatesXmlRenderer.java:101)
    at org.codehaus.mojo.versions.DependencyUpdatesReport.doGenerateReport (DependencyUpdatesReport.java:177)
```

Minimal test-pom (run `mvn site` on it):
```xml
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <groupId>test</groupId>
  <artifactId>test</artifactId>
  <version>1-SNAPSHOT</version>
  <packaging>pom</packaging>
  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>org.codehaus.mojo</groupId>
        <artifactId>versions-maven-plugin</artifactId>
        <version>[2.6.0-SNAPSHOT,2.99)</version>
      </dependency>
    </dependencies>
  </dependencyManagement>
  <build>
    <pluginManagement>
      <plugins>
        <plugin>
          <artifactId>maven-site-plugin</artifactId>
          <version>3.7.1</version>
        </plugin>
        <plugin>
          <groupId>org.codehaus.mojo</groupId>
          <artifactId>versions-maven-plugin</artifactId>
          <version>2.7</version>
          <configuration>
            <formats>
              <format>html</format>
              <format>xml</format>
            </formats>
          </configuration>
        </plugin>
      </plugins>
    </pluginManagement>
  </build>
  <reporting>
    <plugins>
      <plugin>
        <groupId>org.codehaus.mojo</groupId>
        <artifactId>versions-maven-plugin</artifactId>
      </plugin>
    </plugins>
  </reporting>
</project>
```

**Comment by ajarmoniuk** [comment]
Should already be fixed. Please retest with the latest version.

**Comment by ajarmoniuk** [comment]
Fixed by cd941512

**Comment by ajarmoniuk** [comment]
DefaultArtifact::version is set to null in its constructor if the dependency specifies a range and there's no recommended version:

```java
    private void selectVersionFromNewRangeIfAvailable()
    {
        if ( ( versionRange != null ) && ( versionRange.getRecommendedVersion() != null ) )
        {
            selectVersion( versionRange.getRecommendedVersion().toString() );
        }
        else
        {
            version = null;
            baseVersion = null;
        }
    }
```
